### ISSUE TITLE
Interaction with sampler objects and EXT_texture_filter_anisotropic

### ISSUE THREAD

**Issue by juj** [KhronosGroup/WebGL#2006]
In WebGL 2, sampler objects move texture sampling specific state from textures to sampler objects.

The extension https://www.khronos.org/registry/webgl/extensions/EXT_texture_filter_anisotropic/ adds a new texture sampling state variable to textures:

```
const GLenum TEXTURE_MAX_ANISOTROPY_EXT       = 0x84FE;
```

When sampler objects are used, should this state variable be available in the state of the sampler object, i.e. should it be possible to set the following?

```
webGL2context.samplerParameterf(samplerObject, anisoExt.TEXTURE_MAX_ANISOTROPY_EXT, 16.0);
```

Correspondingly, get the following?

```
webGL2context.getSamplerParameter(samplerObject, anisoExt.TEXTURE_MAX_ANISOTROPY_EXT, 'float');
```

Neither on native GLES3 or desktop OpenGL side specifications, I was not able to find a spec wording about this, although testing locally on native desktop OpenGL 4.4.0 (Windows+NVidia), this works. Curiously desktop OpenGL 4.5 core spec pdf does not mention the string "anisotropy" at all.

The wording text for the native side extension at https://www.opengl.org/registry/specs/EXT/texture_filter_anisotropic.txt predates sampler objects by a long time, and it is possible that the spec text was never updated nor a new spec was created to highlight this, but native drivers just went ahead to support it on their own.

**Comment by kenrussell** [comment]
Thanks for reporting this. Anisotropic texture filtering has always been an OpenGL extension to the best of my knowledge, and it's a good point that its interactions with sampler objects may not be well defined even in the native specs.

WebGL's wrapper for EXT_texture_filter_anisotropic should define the interactions with WebGL 2.0, and include a conformance test. (This will require WebGL 2.0 implementations to be updated, of course.)

If you have the time to contribute a test under sdk/tests/conformance2/extensions/ which asserts the behavior (and I think it should be assumed that anisotropy is one of the properties contained in the sampler object), that would be greatly appreciated. Presumably this can be emulated reasonably easily in WebGL 2.0 implementations if it turns out that some drivers don't implement it properly.

**Comment by Grief** [comment]
Here is the code:

        let gl = canvas.getContext("webgl2");
        let extAnisotropic = gl.getExtension('EXT_texture_filter_anisotropic');
        let sampler = gl.createSampler();
        gl.samplerParameterf(sampler, extAnisotropic.TEXTURE_MAX_ANISOTROPY_EXT, 16.0);

I suppose it should look like that because of the following statement in [wiki](https://www.khronos.org/opengl/wiki/Sampler_Object#Anisotropic_filtering):
>To use anisotropic filtering, set the GL_TEXTURE_MAX_ANISOTROPY_EXT sampling parameter. This parameter is floating-point, and can be set between 1.0f and an implementation-defined maximum anisotropy (queried with GL_MAX_TEXTURE_MAX_ANISOTROPY_EXT). Any value greater than 1.0f counts as a use of anisotropic filtering.

But at least in Chrome it fails with

    WebGL: INVALID_ENUM: samplerParameter: invalid parameter name

**Comment by jdashg** [comment]
Fixed in #2925.
